Tools/build/generate_token.py

```diff
@@ -108,14 +107,14 @@ def make_h(infile, outfile='Include/internal/pycore_token.h'):
 
     defines = []
     for value, name in enumerate(tok_names[:ERRORTOKEN + 1]):
-        defines.append("#define %-15s %d\n" % (name, value))
+        defines.append("#define %-15s %d\n" % (name, value))  # noqa: UP031
```

Comment:
Why noqa UP031 in this file?

Reply:
I think it's more readable with a `%`-style rather than with `{}` (especially when writing C code)

Reply:
I'd consider using the `per-file-ignores` setting to exclude the whole file from UP031 if you're not a fan of it, rather than having so many `noqa` comments in this file. Or you could do `# ruff: noqa: UP031` near the top of the file, which would have the same effect

Reply:
Ok, let's do that. Smaller diff and the maintainer can decide what do to later. Considering that UP031 is responsible for most of the diff, I'll also disable it elsewhere. It's better that way I think.